Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,267 @@
+        {
+            return int.TryParse(input = PromptInput(message), out result);
+        }
+
+        /// <summary>
+        /// Prompts for a float value. Converts the string input to a float if possible.
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out float result, out string input)
+        {
+            return float.TryParse(input = PromptInput(message), NumberStyles.Float,
+                CultureInfo.InvariantCulture, out result);
+        }
+
+        /// <summary>
+        /// Checks if the given name is valid.
+        /// </summary>
+        /// <param name="name">The name to check.</param>
+        /// <param name="result">The resulting string of the check. Is "-" if invalid.</param>
+        /// <returns>Returns true if name is not empty or null, false otherwise.</returns>
+        private static bool IsValidName(string name, out string result)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                Console.Write("Name cannot be empty.");
+                result = EmptyName;
+                return false;
+            }
+
+            result = name;
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if the given value is valid.
+        /// </summary>
+        /// <param name="value">The value to check.</param>
+        /// <param name="infimum">The exclusive infimum. Use flaot.MaxValue to set no lower limit.</param>
+        /// <param name="supremum">The exclusive maximum. Use flaot.MaxValue to set no upper limit.</param>
+        /// <returns></returns>
+        private static bool IsValidInt(int value, int infimum = int.MaxValue, int supremum = int.MinValue)
+        {
+            bool hasNoLowerLimit = infimum == int.MaxValue;
+            bool hasNoUpperLimit = supremum == int.MinValue;
+            return (hasNoLowerLimit || value > infimum) && (hasNoUpperLimit || value < supremum);
+        }
+
+        /// <summary>
+        /// Checks if the given value is valid.
+        /// </summary>
+        /// <param name="value">The value to check.</param>
+        /// <param name="infimum">The exclusive infimum. Use flaot.MaxValue to set no lower limit.</param>
+        /// <param name="supremum">The exclusive maximum. Use flaot.MaxValue to set no upper limit.</param>
+        /// <returns>True if the value is valid, false otherwise.</returns>
+        private static bool IsValidFloat(float value, float infimum = float.MaxValue, float supremum = float.MinValue)
+        {
+            bool hasNoLowerLimit = infimum == float.MaxValue;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885761)